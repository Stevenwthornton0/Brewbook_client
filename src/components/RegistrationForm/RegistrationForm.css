.RegistrationForm {
    width: 250px;
    height: 450px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: 15px;
}

label {
    display: block;
    font-size: 14px;
    font-weight: 500;
    margin-bottom: 4px;
} 
