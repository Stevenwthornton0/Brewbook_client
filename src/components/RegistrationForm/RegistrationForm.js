import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import LoaderSpinner from '../../Utils/LoaderSpinner';
import { Input, Button } from '../../Utils/Utils';
import './RegistrationForm.css';

export default class RegistrationForm extends Component {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = { 
        error: null,
        waiting: false 
    };

    
    toSnakeCase = function(string) {
      var s;
      s = string.replace(/[^\w\s]/g, "");
      s = s.replace(/\s+/g, " ");
      return s.toLowerCase().split(' ').join('_');
    };

    handleSubmit = ev => {
        ev.preventDefault()
        const { first_name, last_name, user_name, password } = ev.target;

        this.setState({ error: null, waiting: true });
        AuthApiService.postUser({
            first_name: first_name.value,
            last_name: last_name.value,
            user_name: user_name.value,
            password: password.value,
            admin: 'false'
        })
            .then(user => {
                first_name.value = ''
                last_name.value = ''
                user_name.value = ''
                password.value = ''
                this.props.onRegistrationSuccess()
            })
            .catch(res => {
                this.setState({error: res.error })
            })
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className='RegistrationForm'
                onSubmit={this.handleSubmit}
            >
                <div role='alert'>
                    {error && <p>{error}</p>}
                </div>
                <div className='first_name'>
                    <label htmlFor='RegistrationForm_first-name'>
                        First Name
                    </label>
                    <Input
                        name='first_name'
                        type='text'
                        required
                        id='RegistrationForm_first-name'
                    />
                </div>
                <div className='last_name'>
                    <label htmlFor='RegistrationForm_last-name'>
                        Last Name
                    </label>
                    <Input
                        name='last_name'
                        type='text'
                        required
                        id='RegistrationForm_last-name'
                    />
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm_user-name'>
                        User Name
                    </label>
                    <Input
                        name='user_name'
                        type='text'
                        required
                        id='RegistrationForm_user-name'
                    />
                </div>
                <div className='password'>
                    <label htmlFor='RegistrationForm_password'>
                        Password
                    </label>
                    <Input
                        name='password'
                        type='password'
                        required
                        id='RegistrationForm_password'
                    />
                </div>
                <div className='requires'>
                    <p className='requires'>Requirements:</p>
                    <p>&bull; an uppercase letter</p>
                    <p>&bull; a lowercase letter</p>
                    <p>&bull; a number</p>
                    <p>&bull; a special character</p>
                </div>
                <Button type='submit'>
                    Register
                </Button>
                <div className='loaderContainer'>
                    {this.state.waiting ? <LoaderSpinner /> : null}
                </div>
            </form>
        )
    }
}