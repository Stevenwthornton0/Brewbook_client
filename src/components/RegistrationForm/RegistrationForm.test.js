import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import RegistrationForm from './RegistrationForm';

Enzyme.configure({ adapter: new Adapter() })

describe(`RegistrationForm smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<RegistrationForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})