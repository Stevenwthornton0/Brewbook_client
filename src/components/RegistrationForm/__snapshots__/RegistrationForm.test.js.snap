// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm smoke test renders without crashing 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="first_name"
  >
    <label
      htmlFor="RegistrationForm_first-name"
    >
      First Name
    </label>
    <Input
      id="RegistrationForm_first-name"
      name="first_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="last_name"
  >
    <label
      htmlFor="RegistrationForm_last-name"
    >
      Last Name
    </label>
    <Input
      id="RegistrationForm_last-name"
      name="last_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm_user-name"
    >
      User Name
    </label>
    <Input
      id="RegistrationForm_user-name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm_password"
    >
      Password
    </label>
    <Input
      id="RegistrationForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="requires"
  >
    <p
      className="requires"
    >
      Requirements:
    </p>
    <p>
      • an uppercase letter
    </p>
    <p>
      • a lowercase letter
    </p>
    <p>
      • a number
    </p>
    <p>
      • a special character
    </p>
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
  <div
    className="loaderContainer"
  />
</form>
`;
